### ISSUE TITLE
on('notification') not working in foreground (iOS)

### ISSUE THREAD

**Issue by MrHakala** [phonegap/phonegap-plugin-push#2783]
solution for me was to use payload in this format
{
  "to" : "bk3RNwTe3H0:CI2k_HHwgIpoDKCIZvvDMExUdFQ3P1...",
  "notification": {},
  "data": {},
  "content_available": true  
}
and add code from old version which trigger notification event when app is in foreground
file  from IOS project, Plugins/AppDelegate+notification.m
in func didReceiveRemoteNotification
added this code from 'else':
``` 
if (application.applicationState != UIApplicationStateActive) {
...
} else {
       NSLog(@"app active");
        PushPlugin *pushHandler = [self getCommandInstance:@"PushNotification"];
        pushHandler.notificationMessage = userInfo;
        pushHandler.isInline = YES;
        [pushHandler notificationReceived];
        
        completionHandler(UIBackgroundFetchResultNewData);
}

_Originally posted by @duca14036 in https://github.com/phonegap/phonegap-plugin-push/issues/2724#issuecomment-474882117_

**Comment by fincha** [comment]
was this merged?

**Comment by frank10gm** [comment]
is this merged?

**Comment by gabn88** [comment]
Not merged sadly. Ios will do nothing if it receives a notification when the app is in the foreground. Spend two days tracking this issue and changing serverside request to find this issue here.

**Comment by duca14036** [comment]
**FCM Payload**
```
[to] => cftX5A2a.....
[notification] => Array
    (
        [title] => Alert
        [body] => Test Message 1
        [force-start] => 1
        [notId] => 2
        [sound] => default
    )
[data] => Array
    (
        [data1] => value1
        [data2] => value2
    )
[content_available] => 1
```
and ios changes work as expected, receiving notification in background and foreground

**Comment by gabn88** [comment]
Yes, it works. But not with the phonegap-plugin-push version that is installed using Cordova plugin add phonegap-plugin-push. You need to modify that file manually.

**Comment by duca14036** [comment]
@gabn88 you can try this repository https://github.com/zetekyinc/phonegap-plugin-push

**Comment by gabn88** [comment]
Thank you @duca14036! That is fast support! I like your fix, because it is very concise and almost without duplicate code!

I ended up using this https://github.com/phonegap/phonegap-plugin-push/issues/2824#issuecomment-540505990 version of the 'fix'. Just because it was the first one that was working for me. I'm unaware of any practical differences with your fix.

Full disclaimer: I also started to look for other libraries, like https://github.com/dpa99c/cordova-plugin-firebasex, because this plugin seems to be not so well maintained. But to be honoust, this fix solves for me all the problems for now.

**Comment by souvickcse** [comment]
@duca14036 
Can you please tell me how to install this npm from this repo(https://github.com/zetekyinc/phonegap-plugin-push) (I am very new in this)

**Comment by Brendan92** [comment]
Just to be clear so the solution is to install this https://github.com/zetekyinc/phonegap-plugin-push? No progress on updating the original plugin?

**Comment by Sugamsm** [comment]
Hello, I added a small fix which displayed the notification on App Foreground status too. Please try:
https://github.com/Sugamsm/phonegap-plugin-push
